### ISSUE TITLE
Stockfish analysis sometimes not shown

### ISSUE THREAD

**Issue by dsjoerg** [veloce/lichobile#1068]
**I'm submitting a ...**  (keep the relevant one):
- bug report

# Bug report

**Current behavior:**

* Play a game
* Go to analysis
* In some but not all positions (perhaps where there is check or mate is imminent), rather than engine analysis all you will see is the hourglass (see attached screenshots)
![IMG_8647](https://user-images.githubusercontent.com/783273/67164346-99cbab00-f347-11e9-92ea-d63235282938.PNG)
![IMG_8617](https://user-images.githubusercontent.com/783273/67164347-99cbab00-f347-11e9-9533-4e41260d1dd6.PNG)

**Expected behavior:**

* In all positions, the stockfish analysis should be shown after some amount of time.  The hourglass should not stay forever.

**Steps to reproduce:**

* Follow instructions above, or
* go to this game: https://lichess.org/zqnhvuOu/black in the app, go to move 76 for white in Analysis, and go to the "gear" tab.

**Environment information**

- phone hardware: iPhone SE
- phone operating system version: 12.3.1
- version of lichess mobile: v6.3.0

**Other information:**

Thank you!

**Comment by amitdash291** [comment]
This issue happens on the Android app too.
Device - One Plus 6T

**Comment by oakkitten** [comment]
same on xiaomi mi a2, the browser version seems to be working fine. the game appears to be always stuck when the king has to move, even when the king is checked by the knight and has only one valid move

**Comment by 18Dylan18** [comment]
Yes, the problem seems to be: when you have N engine lines enabled (i.e. 3 for instance) and there exist fewer than N possible moves (or possible lines)
(I.e. you're in check and there are only 2 possible moves)

This should be fairly easy to fix, but I don't have the coding background to know how to do so.
